lib/SILOptimizer/Utils/ConstExpr.cpp

```diff
@@ -539,6 +739,21 @@ ConstExprFunctionState::evaluateFlowSensitive(SILInstruction *inst) {
       isa<StrongReleaseInst>(inst))
     return None;
 
+  // If this is a special flow-sensitive instruction like a stack allocation,
+  // store, copy_addr, etc, we handle it specially here.
+  if (auto asi = dyn_cast<AllocStackInst>(inst)) {
+    createMemoryObject(asi, SymbolicValue::getUninitMemory());
+    return None;
+  }
+
+  // If this is a deallocation of a memory object that we may be tracking,
+  // remove the memory from the set.  We don't *have* to do this, but it seems
+  // useful for hygiene.
```

Comment:
Is there some kind of assertion that could be added to verify this hygiene? For example, at function exit?

Reply:
Now that my attention has been drawn to this code-comment, I realize that the code-comment and the code is wrong. It's left over from an earlier memory implementation where the memory values actually got stored in `calculatedValues`.

I have changed the code to do nothing here, and updated the comment to explain. We could do something a bit more elaborate, like add a new "freed" state to the memory and make it an assertion failure whenever you try to do any operations to "freed" memory.